argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -48,6 +48,9 @@ def to_datasets(records: List["Record"]) -> HFDataset:
         """
         record_dicts = GenericIO.to_dict(records, flatten=True)
         dataset = HFDataset.from_dict(record_dicts)
+        image_fields = HFDatasetsIO._get_image_fields(schema=schema)
+        if image_fields:
+            dataset = HFDatasetsIO._cast_uris_as_images(hf_dataset=dataset, columns=image_fields)
```

Comment:
I feel like that would be too many short methods. I think that the class need to be refactored when we bring custom and chat field in, so that there's a methods for casting features.